framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -63,4 +91,99 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			await this._sidechainAPIClient.disconnect();
 		}
 	}
+	public async getCertificateFromAggregateCommit(
+		aggregateCommit: AggregateCommit,
+	): Promise<Certificate> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
+			height: aggregateCommit.height,
+		});
+
+		const certificate = computeCertificateFromBlockHeader(block.header);
+		certificate.aggregationBits = block.header.aggregateCommit.aggregationBits;
+		certificate.signature = aggregateCommit.certificateSignature;
+
+		return certificate;
+	}
+
+	public async getNextCertificateFromAggregateCommits(
+		lastCertifiedHeight: number,
+		aggregateCommits: AggregateCommit[],
+	): Promise<Certificate | null> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
+			height: lastCertifiedHeight,
+		});
+
+		const validatorsData = this._chainConnectorDB.getChainConnectorInfo().validatorsHashPreimage[
+			block.header.validatorsHash!.toString('hex')
+		];
+
+		const blsKeyToBFTWeight: Record<string, BigInt> = {};
+
+		validatorsData.validators.forEach((validator: BFTValidator) => {
+			blsKeyToBFTWeight[validator.blsKey.toString('hex')] = validator.bftWeight;
+		});
+
+		const bftHeights = await this._mainchainAPIClient.invoke<BFTHeights>('consensus_getBFTHeights');
+
+		let height = bftHeights.maxHeightCertified;
+
+		while (height > lastCertifiedHeight) {
+			if (aggregateCommits[height] !== undefined) {
+				const valid = await this.checkChainOfTrust(
+					block.header.validatorsHash!,
+					blsKeyToBFTWeight,
+					validatorsData.certificateThreshold,
+					aggregateCommits[height],
+				);
+
+				if (valid) {
+					return this.getCertificateFromAggregateCommit(aggregateCommits[height]);
+				}
+			}
+
+			height -= 1;
+		}
+
+		return null;
+	}
+
+	public async checkChainOfTrust(
+		lastValidatorsHash: Buffer,
+		blsKeyToBFTWeight: Record<string, BigInt>,
+		lastCertificateThreshold: BigInt,
+		aggregateCommit: AggregateCommit,
+	): Promise<boolean> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
```

Comment:
sidechainAPI should be called here

Reply:
also, here you can get block header from `getChainConnectorInfo` blockheader list